src/transport/SecureSessionMgr.cpp

```diff
@@ -292,12 +317,47 @@ void SecureSessionMgr::OnMessageReceived(const PeerAddress & peerAddress, System
 void SecureSessionMgr::MessageDispatch(const PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
                                        System::PacketBufferHandle && msg)
 {
+    Transport::UnauthenticatedSession * session = mUnauthenticatedSessions.FindOrAllocateEntry(peerAddress);
+    if (session == nullptr)
+    {
+        ChipLogError(Inet, "UnauthenticatedSession exhausted");
+        return;
+    }
+
+    SecureSessionMgrDelegate::DuplicateMessage isDuplicate = SecureSessionMgrDelegate::DuplicateMessage::No;
+
+    // Verify message counter
+    CHIP_ERROR err = session->GetPeerMessageCounter().VerifyOrTrustFirst(packetHeader.GetMessageId());
+    if (err == CHIP_ERROR_DUPLICATE_MESSAGE_RECEIVED)
+    {
+        ChipLogDetail(Inet, "Received a duplicate message with MessageId: %" PRIu32, packetHeader.GetMessageId());
+        isDuplicate = SecureSessionMgrDelegate::DuplicateMessage::Yes;
+        err         = CHIP_NO_ERROR;
+    }
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Inet, "Message counter verify failed, err = %" CHIP_ERROR_FORMAT, err.Format());
```

Comment:
In theory, it should be impossible to hit this line.

I'm not sure if we can use that knowledge for any type of simplification though.

Reply:
Replaced with a `VerifyOrDie`